### PR TITLE
Ensure account can be authorized only within 15 minutes after clicking the oauth button

### PR DESCRIPTION
**Pull request by sgiehl**

### CODE REVIEW

File: Controller.php [hunk 1]
```diff
@@ -115,6 +115,28 @@ public function index($errorMessage = false)
...
+    public function forwardToAuth()
+    {
+        Piwik::checkUserHasSuperUserAccess();
+
+        Nonce::checkNonce('gaimport.auth', Common::getRequestVar('auth_nonce'));
+
+        $session = $this->getSession();
+
+        $session->gaauthtime = time();
+        $session->setExpirationSeconds(60 * 15, 'gaauthtime');
```
Comment (diosmosis) [inline Controller.php]: @sgiehl can you make the same change as in the other PR?

Comment (sgiehl) [reply Controller.php]: @diosmosis done.

**Comment by diosmosis** [comment]
Tested locally and this doesn't seem to work for me... when I authorize, no matter how long I wait to invoke the processAuthCode action, I end up redirected to the dashboard. Not sure why.

**Comment by sgiehl** [comment]
Had the same issue yesterday while testing, but it turned out that the redirect url in the client configuration on google was somehow not including any parameters anymore.

**Comment by diosmosis** [comment]
This wasn't the issue for me, the URL was right, but it stopped for some reason today, so...

**Pull request merged**